filament/src/PostProcessManager.cpp

```diff
@@ -1308,27 +1308,53 @@ FrameGraphId<FrameGraphTexture> PostProcessManager::bloomPass(FrameGraph& fg,
         });
     }
 
+    // TODO: this is a feature flag that should be passed in from above.
+    // Reading from one mip while writing to a different mip is covered in the OpenGL ES 3.0
+    // specification under the heading  "Feedback Loops Between Textures and the Framebuffer".
+    // Some OpenGL implementations do not permit this behavior (e.g. macOS Chrome + WebGL 2.0).
+#if defined(__EMSCRIPTEN__)
+    const bool disableFeedback = true;
+#else
+    const bool disableFeedback = false;
+#endif
+
     struct BloomPassData {
         FrameGraphId<FrameGraphTexture> in;
         FrameGraphId<FrameGraphTexture> out;
+        FrameGraphId<FrameGraphTexture> stage;
         FrameGraphRenderTargetHandle outRT[kMaxBloomLevels];
+        FrameGraphRenderTargetHandle stageRT[kMaxBloomLevels];
     };
 
     // downsample phase
     auto& bloomDownsamplePass = fg.addPass<BloomPassData>("Bloom Downsample",
             [&](FrameGraph::Builder& builder, auto& data) {
                 data.in = builder.sample(input);
-                data.out = builder.createTexture("Bloom Texture", {
+                data.out = builder.createTexture("Bloom Out Texture", {
                         .width = width,
                         .height = height,
                         .levels = bloomOptions.levels,
                         .format = outFormat
                 });
                 data.out = builder.write(builder.sample(data.out));
 
+                if (disableFeedback) {
+                    data.stage = builder.createTexture("Bloom Stage Texture", {
+                            .width = width,
+                            .height = height,
+                            .levels = bloomOptions.levels,
+                            .format = outFormat
+                    });
+                    data.stage = builder.write(builder.sample(data.stage));
+                } else {
+                    data.stage = builder.write(builder.sample(data.out));
+                }
+
                 for (size_t i = 0; i < bloomOptions.levels; i++) {
-                    data.outRT[i] = builder.createRenderTarget("Bloom target", {
+                    data.outRT[i] = builder.createRenderTarget("Bloom Out Target", {
                             .attachments = {{ data.out, uint8_t(i) }} });
```

Comment:
I dont know why it doesn't fail -- but data.out is invalid here in the normal case (see my comment above)